tests/backup/backup_cluster_share_test.go

```diff
@@ -0,0 +1,481 @@
+		})
+
+		Step("Validate the restore object", func() {
+			log.InfoD("validating the restore object")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			restoreInspectRequest := &api.RestoreInspectRequest{
+				Name:  restoreName,
+				OrgId: BackupOrgID,
+				Uid:   restoreUID,
+			}
+			_, err = Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+			log.FailOnError(err, "inspect restore %s", restoreName)
+		})
+
+		Step("Cleanup", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			backupDriver := Inst().Backup
+			backupUID, err := backupDriver.GetBackupUID(ctx, backupName, BackupOrgID)
+			log.FailOnError(err, "Failed while trying to get backup UID for - [%s]", backupName)
+			log.InfoD("Deleting backup")
+			_, err = DeleteBackup(backupName, backupUID, BackupOrgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting backup [%s]", backupName))
+
+			log.InfoD(fmt.Sprintf("Delete restore with name [%s]", restoreName))
+			err = DeleteRestore(restoreName, BackupOrgID, ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Deleting restore [%s]", restoreName))
+		})
+
+	})
+	// This testcase verifies the cluster un-share operation when there is a backupschedule.
+	It("VerifyClusterUnShareWhenBackupSchedulesAreDeleted", func() {
+		//TODO: Need to update the testrail ID
+		StartPxBackupTorpedoTest("VerifyClusterUnShareWhenBackupSchedulesAreDeleted", "VerifyClusterUnShareWhenBackupSchedulesAreDeleted during the cluster unshare", nil, 0, Sgajawada, Q2FY25)
+
+		var (
+			scheduleName               string
+			periodicSchedulePolicyName string
+			periodicSchedulePolicyUid  string
+			clusterUid                 string
+		)
+
+		Step("Get login context for the test users", func() {
+			for i := 1; i <= 2; i++ {
+				userName := testUsers[i].name
+				ctxNonAdmin, err := backup.GetNonAdminCtx(userName, CommonPassword)
+				log.FailOnError(err, "Fetching non admin ctx")
+				user := testUsers[i]
+				user.ctx = ctxNonAdmin
+				testUsers[i] = user
+			}
+		})
+
+		Step("Create a cluster object with User1 and Share with User2", func() {
+			err := AddSourceCluster(testUsers[1].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of source [%s] cluster with %s ctx", SourceClusterName, testUsers[1].name))
+			clusterUid, err = Inst().Backup.GetClusterUID(testUsers[1].ctx, BackupOrgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+
+			// Share the source cluster to testuser2
+			err = ShareCluster(SourceClusterName, nil, []string{testUsers[2].name}, testUsers[1].ctx)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying share of source [%s] cluster with %s user using %s ctx", SourceClusterName, testUsers[2].name, testUsers[1].name))
+		})
+
+		Step("Create backup location and cloud setting with User2", func() {
```

Comment:
This one too we are repeating same lines of code

Reply:
done